addons/mail/static/tests/mail_test_helpers.test.js

```diff
@@ -0,0 +1,118 @@
+import { expect, test } from "@odoo/hoot";
+
+import { defineModels, models, fields, makeKwArgs } from "@web/../tests/web_test_helpers";
+import { startServer, defineMailModels } from "./mail_test_helpers";
+import { mailDataHelpers } from "@mail/../tests/mock_server/mail_mock_server";
+
+class TestModel extends models.Model {
+    _name = "test.model";
+
+    name = fields.Char({ string: "Name" });
+    other = fields.One2many({
+        relation: "test.model2",
+        string: "Other",
+    });
+
+    _to_store(store, fields) {
+        store._add_record_fields(
+            this,
+            fields.filter((f) => f !== "name")
+        );
+        for (const model of this) {
+            if (fields.includes("name")) {
+                store._add_record_fields(this.browse(model.id), {
+                    name: model.name,
+                    test: "test",
+                });
+            }
+        }
+    }
+
+    get _to_store_defaults() {
+        return [
+            "name",
+            mailDataHelpers.Store.many(
+                "other",
+                makeKwArgs({
+                    fields: ["name"],
+                })
+            ),
+        ];
+    }
+}
+
+class TestModel2 extends models.Model {
+    _name = "test.model2";
+
+    name = fields.Char({ string: "Name" });
+    first = fields.Many2one({
+        relation: "test.model",
+        string: "First",
+    });
+}
```

Comment:
Wouldn't it be possible to test with real models instead?

Also the value of these test is rather low though, if we have an issue with to store many other tests would be broken.
Unlike for the real server where the tests are there to ensure the code is correct, the goal of the mock server is to support the other tests, and if the tests are fine it means the mock server can be considered fine.